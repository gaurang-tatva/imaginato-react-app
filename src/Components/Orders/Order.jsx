import React from "react";
import {
  Button,
  Typography,
  IconButton,
  Card,
  Box,
  makeStyles
} from "@material-ui/core";
import LocationOnIcon from "@material-ui/icons/LocationOn";
import NoteIcon from "@material-ui/icons/Note";
import AttachMoneyIcon from "@material-ui/icons/AttachMoney";
import PropTypes from "prop-types";

const useStyles = makeStyles(theme => ({
  root: {
    width: "100%"
  },
  iconContainer: {
    padding: theme.spacing(1)
  },
  orderNameText: {
    paddingTop: theme.spacing(1),
    paddingLeft: theme.spacing(1),
    margin: 0,
    whiteSpace: "nowrap",
    overflow: "hidden",
    textOverflow: "ellipsis"
  },
  textContainer: {
    display: "flex",
    alignItems: "flex-start",
    justifyContent: "space-between"
  },
  itemContainer: {
    display: "flex",
    alignItems: "center",
    justifyContent: "flex-start"
  },
  priceContainer: {
    display: "flex",
    alignItems: "center",
    justifyContent: "flex-start",
    padding: theme.spacing(0, 2)
  },
  priceIcon: {
    padding: theme.spacing(1, 0),
    "> svg": {
      fontSize: "18px"
    }
  },
  buttonContainer: {
    padding: theme.spacing(1),
    textAlign: "center"
  },
  ellipsis: {
    whiteSpace: "nowrap",
    overflow: "hidden",
    textOverflow: "ellipsis"
  },
  editButton: {
    margin: theme.spacing(0, 1)
  }
}));

/**
 * Order component
 */

function Order({ orderData, handleViewMore, editOrderModal }) {
  const classes = useStyles();

  return (
    <Card className={classes.root}>
      <Box>
        <Box className={classes.textContainer}>
          <Typography variant="body1" className={classes.orderNameText}>
            <strong>{orderData.name && orderData.name}</strong>
          </Typography>
          <div className={classes.priceContainer}>
            <IconButton color="primary" className={classes.priceIcon}>
              <AttachMoneyIcon />
            </IconButton>
            <Typography variant="body1" className={classes.ellipsis}>
              <strong>{orderData.price && orderData.price}</strong>
            </Typography>
          </div>
        </Box>
        <Box>
          <div className={classes.itemContainer}>
            <IconButton color="primary" className={classes.iconContainer}>
              <LocationOnIcon />
            </IconButton>
            <Typography variant="body1" className={classes.ellipsis}>
              {orderData.location && orderData.location}
            </Typography>
          </div>
          <div className={classes.itemContainer}>
            <IconButton color="primary" className={classes.iconContainer}>
              <NoteIcon />
            </IconButton>
            <Typography variant="body1" className={classes.ellipsis}>
              {orderData.note ? orderData.note : "---"}
            </Typography>
          </div>
          <div className={classes.buttonContainer}>
            <Button
              variant="contained"
              color="primary"
              onClick={() => handleViewMore(orderData)}
            >
              View
            </Button>
            <Button
              variant="contained"
              color="secondary"
              onClick={() => editOrderModal(orderData)}
              className={classes.editButton}
            >
              Edit
            </Button>
          </div>
        </Box>
      </Box>
    </Card>
  );
}

Order.propTypes = {
  orderData: PropTypes.object,
  handleViewMore: PropTypes.func
};

export default Order;
