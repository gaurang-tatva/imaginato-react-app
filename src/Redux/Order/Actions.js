export const ACTION_TYPE = {
  FETCH_ORDERS: "FETCH_ORDERS",
  ADD_ORDER: "ADD_ORDER",
  EDIT_ORDER: "EDIT_ORDER"
};
