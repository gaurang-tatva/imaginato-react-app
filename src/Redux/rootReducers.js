import Order from "./Order/Reducers";
import { combineReducers } from "redux";

const rootReducer = combineReducers({
  Order
});

export default rootReducer;
