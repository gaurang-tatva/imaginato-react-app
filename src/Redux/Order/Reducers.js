import { ACTION_TYPE } from "./Actions";

const initialState = {
  orderList: [
    {
      id: "TXT_1",
      name: "Orange Juice",
      price: 15,
      location: "Beijing, China",
      note: "Please add tissue."
    },
    {
      id: "TXT_2",
      name: "Apple Juice",
      price: 15,
      location: "Shanghai, China",
      note: "Please add extra glass"
    },
    {
      id: "TXT_3",
      name: "Banana Shake",
      price: 20,
      location: "Tianjin, China",
      note: "Please add extra spoon."
    },
    {
      id: "TXT_4",
      name: "Carrot Juice",
      price: 15,
      location: "Beijing, China",
      note: "Please add tissue."
    },
    {
      id: "TXT_5",
      name: "Cherry Juice",
      price: 15,
      location: "Shanghai, China",
      note: "Please add extra glass"
    },
    {
      id: "TXT_6",
      name: "Grape Juice",
      price: 20,
      location: "Tianjin, China",
      note: "Please add extra spoon."
    }
  ]
};

/**
 * Order Reducer
 */
const OrderReducer = (state = initialState, action) => {
  switch (action.type) {
    case ACTION_TYPE.FETCH_ORDERS:
      return {
        ...state,
        orderList: action.orderList
      };
    case ACTION_TYPE.ADD_ORDER:
      return {
        ...state,
        orderList: action.orderList
      };
    case ACTION_TYPE.EDIT_ORDER:
      return {
        ...state,
        orderList: action.orderList
      };
    default:
      return state;
  }
};

export default OrderReducer;
