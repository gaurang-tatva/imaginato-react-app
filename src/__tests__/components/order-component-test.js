import React from "react";
import { Provider } from "react-redux";
import Order from "../../Components/Orders/Order";
import renderer from "react-test-renderer";
import { store } from "../../Redux/Store";
import orderData from './../../fixture/orders.json'

describe("Order Component", () => {
  it("should render without throwing an error", async () => {
    const rendered = renderer.create(
      <Provider dispatch={jest.fn} store={store}>
        <Order  orderData={orderData} handleViewMore={jest.fn} />
      </Provider>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
