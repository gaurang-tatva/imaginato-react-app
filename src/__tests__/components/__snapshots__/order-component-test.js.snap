// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Component should render without throwing an error 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiBox-root MuiBox-root-41"
  >
    <div
      className="MuiBox-root MuiBox-root-42 makeStyles-textContainer-4"
    >
      <p
        className="MuiTypography-root makeStyles-orderNameText-3 MuiTypography-body1"
      >
        <strong>
          Orange Juice
        </strong>
      </p>
      <div
        className="makeStyles-priceContainer-6"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root makeStyles-priceIcon-7 MuiIconButton-colorPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.8 10.9c-2.27-.59-3-1.2-3-2.15 0-1.09 1.01-1.85 2.7-1.85 1.78 0 2.44.85 2.5 2.1h2.21c-.07-1.72-1.12-3.3-3.21-3.81V3h-3v2.16c-1.94.42-3.5 1.68-3.5 3.61 0 2.31 1.91 3.46 4.7 4.13 2.5.6 3 1.48 3 2.41 0 .69-.49 1.79-2.7 1.79-2.06 0-2.87-.92-2.98-2.1h-2.2c.12 2.19 1.76 3.42 3.68 3.83V21h3v-2.15c1.95-.37 3.5-1.5 3.5-3.55 0-2.84-2.43-3.81-4.7-4.4z"
              />
            </svg>
          </span>
        </button>
        <p
          className="MuiTypography-root makeStyles-ellipsis-9 MuiTypography-body1"
        >
          <strong>
            15
          </strong>
        </p>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-94"
    >
      <div
        className="makeStyles-itemContainer-5"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root makeStyles-iconContainer-2 MuiIconButton-colorPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
              />
            </svg>
          </span>
        </button>
        <p
          className="MuiTypography-root makeStyles-ellipsis-9 MuiTypography-body1"
        >
          Beijing, China
        </p>
      </div>
      <div
        className="makeStyles-itemContainer-5"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root makeStyles-iconContainer-2 MuiIconButton-colorPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 10l-6-6H4c-1.1 0-2 .9-2 2v12.01c0 1.1.9 1.99 2 1.99l16-.01c1.1 0 2-.89 2-1.99v-8zm-7-4.5l5.5 5.5H15V5.5z"
              />
            </svg>
          </span>
        </button>
        <p
          className="MuiTypography-root makeStyles-ellipsis-9 MuiTypography-body1"
        >
          Please add tissue.
        </p>
      </div>
      <div
        className="makeStyles-buttonContainer-8"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            View
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-editButton-10 MuiButton-containedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Edit
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
